server/src/http/broadcaster.rs

```diff
@@ -194,9 +201,16 @@ impl Broadcaster {
     }
 
     /// Broadcast new features to all clients.
-    pub async fn broadcast(&self) {
+    pub async fn broadcast(&self, environment: Option<String>) {
         let mut client_events = Vec::new();
-        for entry in self.active_connections.iter() {
+
+        for entry in self.active_connections.iter().filter(|entry| {
+            if let Some(env) = &environment {
+                entry.key().environment == *env
```

Comment:
This line fails for now because it expects the refactor in #618 to go through first. I'd like to leave this as is for now, because the extra logic required in the old setup is a little convoluted. However, it means build/tests will fail until the updated is in.